relayer/band/client.go

```diff
@@ -84,22 +102,41 @@ func (c *client) Connect(timeout uint) error {
 			continue // Try the next endpoint if starting the client fails
 		}
 
-		// Create a new client context and configure it with necessary parameters
-		encodingConfig := MakeEncodingConfig()
-		ctx := cosmosclient.Context{}.
-			WithClient(client).
-			WithCodec(encodingConfig.Marshaler).
-			WithInterfaceRegistry(encodingConfig.InterfaceRegistry)
-
-		c.Context = ctx
-		c.QueryClient = NewQueryClient(tunneltypes.NewQueryClient(ctx), bandtsstypes.NewQueryClient(ctx))
+		c.Context.Client = client
+		c.Context.NodeURI = rpcEndpoint
+		c.QueryClient = NewQueryClient(tunneltypes.NewQueryClient(c.Context), bandtsstypes.NewQueryClient(c.Context))
 
 		c.Log.Info("Connected to Band chain", zap.String("endpoint", rpcEndpoint))
 
 		return nil
 	}
+	return fmt.Errorf("no available RPC endpoint")
+}
 
-	return nil
+// StartLivelinessCheck starts the liveliness check for the BandChain.
+func (c *client) startLivelinessCheck(ctx context.Context, timeout uint, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	for {
+		select {
+		case <-ctx.Done():
+			c.Log.Info("Stopping liveliness check")
+
+			ticker.Stop()
+
+			return
+		case <-ticker.C:
+			if _, err := c.Context.Client.Status(ctx); err != nil {
+				c.Log.Error(
+					"BandChain client disconnected",
+					zap.String("rpcEndpoint", c.Context.NodeURI),
+					zap.Error(err),
+				)
+				if err := c.connect(timeout); err != nil {
+					c.Log.Error("Liveliness check: unable to reconnect to any endpoints", zap.Error(err))
```

Comment:
Selected rpc from startLivenessCheck should be active. 
- suggest solution check status inside the connect function if required.